movement/watch_faces/complication/breathing_face.c

```diff
@@ -0,0 +1,198 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include "breathing_face.h"
+#include "watch.h"
+
+void breathing_face_setup(movement_settings_t *settings, uint8_t watch_face_index, void ** context_ptr) {
+    // These next two lines just silence the compiler warnings associated with unused parameters.
+    // We have no use for the settings or the watch_face_index, so we make that explicit here.
+    (void) settings;
+    (void) watch_face_index;
+    // At boot, context_ptr will be NULL indicating that we don't have anyplace to store our context.
+    if (*context_ptr == NULL) {
+        // in this case, we allocate an area of memory sufficient to store the stuff we need to track.
+        *context_ptr = malloc(sizeof(breathing_state_t));
+    }
+}
+
+void breathing_face_activate(movement_settings_t *settings, void *context) {
+    // same as above: silence the warning, we don't need to check the settings.
+    (void) settings;
+    // we do however need to set some things in our context. Here we cast it to the correct type...
+    breathing_state_t *state = (breathing_state_t *)context;
+    // ...and set the initial state of our watch face.
+    state->current_stage = 0;
+    state->sound_on = true;
+}
+
+const int NOTE_LENGTH = 80;
+
+void beep_in (void) {
+        const BuzzerNote notes[] = {
+            BUZZER_NOTE_C4,
+            BUZZER_NOTE_D4,
+            BUZZER_NOTE_E4,
+        };
+        const uint16_t durations[] = {
+            NOTE_LENGTH,
+            NOTE_LENGTH,
+            NOTE_LENGTH
+        };
+        for(size_t i = 0, count = sizeof(notes) / sizeof(notes[0]); i < count; i++) {
+            watch_buzzer_play_note(notes[i], durations[i]);
+        }
+}
+
+void beep_in_hold (void) {
+        const BuzzerNote notes[] = {
+            BUZZER_NOTE_E4,
+            BUZZER_NOTE_REST,
+            BUZZER_NOTE_E4,
+        };
+        const uint16_t durations[] = {
+            NOTE_LENGTH,
+            NOTE_LENGTH * 2,
+            NOTE_LENGTH,
+        };
+        for(size_t i = 0, count = sizeof(notes) / sizeof(notes[0]); i < count; i++) {
+            watch_buzzer_play_note(notes[i], durations[i]);
+        }
+}
+
+void beep_out (void) {
+        const BuzzerNote notes[] = {
+            BUZZER_NOTE_E4,
+            BUZZER_NOTE_D4,
+            BUZZER_NOTE_C4,
+        };
+        const uint16_t durations[] = {
+            NOTE_LENGTH,
+            NOTE_LENGTH,
+            NOTE_LENGTH,
+        };
+        for(size_t i = 0, count = sizeof(notes) / sizeof(notes[0]); i < count; i++) {
+            watch_buzzer_play_note(notes[i], durations[i]);
+        }
+}
+
+void beep_out_hold (void) {
+        const BuzzerNote notes[] = {
+            BUZZER_NOTE_C4,
+            BUZZER_NOTE_REST * 2,
```

Comment:
I think this `*2` should be applied to `NOTE_LENGTH` instead.

Reply:
Yes. Good catch. Thanks.